AstroFaker replacing AstroData cause test to fail

We all know that if we import `astrofaker` after we import `astrodata`, our objects will become `AstroFaker` instances. Up to now, this hasn't been a problem. The recent GeminiDRSoftware/DRAGONS#78 PR introduced a test that suffered the consequences of this behavior. See the traceback bellow:

```
ad = <astrofaker.gnirs.AstroFakerGnirs object at 0x7fc087138e50>
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7fc087c1e6d0>

    @pytest.mark.dragons_remote_data
    @pytest.mark.parametrize("ad", ["N20190101S0102.fits"], indirect=True)
    def test_ra_and_dec_wcs_fallback(ad, monkeypatch):
        """
        Tests that the get the RA/DEC coordinates using descriptors when the WCS coordinate mapping fails
    
        Parameters
        ----------
        ad : fixture
            Custom fixture that downloads and opens the input file.
        """
        def _fake_wcs_call():
            return None
    
        monkeypatch.setattr(ad, 'wcs_ra', _fake_wcs_call)
        monkeypatch.setattr(ad, 'wcs_dec', _fake_wcs_call)
>       assert(ad.ra() == ad.phu.get('RA', None))

gemini_instruments/gnirs/tests/test_gnirs.py:156: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <astrofaker.gnirs.AstroFakerGnirs object at 0x7fc087138e50>

    @astro_data_descriptor
    def ra(self):
        """
        Returns the Right Ascension of the center of the field in degrees.
        Uses the RA derived from the WCS, unless it is wildly different from
        the target RA stored in the headers (with telescope offset and in
        ICRS).  When that's the case the target RA is used.
    
        Returns
        -------
        float
            Right Ascension of the target in degrees.
        """
        # In general, the GNIRS WCS is the way to go. But sometimes the DC
        # has a bit of a senior moment and the WCS is miles off (presumably
        # still has values from the previous observation or something.
        # Who knows.  So we do a sanity check on it and use the target values
        # if it's messed up
    
        wcs_ra = self.wcs_ra()
        if wcs_ra is None:
            return self.phu.get('RA', None)
        try:
            tgt_ra = self.target_ra(offset=True, icrs=True)
        except:  # Return WCS value if we can't get our sanity check
            return wcs_ra
>       delta = abs(wcs_ra - tgt_ra)
E       TypeError: unsupported operand type(s) for -: 'function' and 'float'

gemini_instruments/gnirs/adclass.py:440: TypeError
```
A quick workaround was to encapsulate `astrofaker` within a fixture (See commit GeminiDRSoftware/DRAGONS@ef8377c60470f7743ae6b7b98102cb242e2eea15), but we might want to keep track of this behavior and change it in the future.

--- a/astrofaker/astrofaker.py
+++ b/astrofaker/astrofaker.py
@@ -143,7 +143,13 @@ class AstroFaker(with_metaclass(abc.ABCMeta, object)):
 
         def override_descriptor(name):
             def fn(self, *args, **kwargs):
-                return self._descriptor_dict[name]
+                value = self._descriptor_dict[name]
+                if callable(value):
+                    return value()
+                else:
+                    return value
+
+            fn.descriptor_method = True
 
             # The first one is py2, the second py3
             try:

--- a/astrofaker/tests/test_gmos.py
+++ b/astrofaker/tests/test_gmos.py
@@ -81,5 +81,16 @@ def test_can_update_descriptor_dispersion_axis_of_astrodata_extensions():
         assert ext.dispersion_axis() == 1
 
 
+def test_override_descriptor():
+    ad = astrofaker.create('GMOS-S')
+    assert ad.wcs_ra() is None
+
+    ad.wcs_ra = 5
+    assert ad.wcs_ra() == 5
+
+    ad.wcs_ra = lambda: 5
+    assert ad.wcs_ra() == 5
+
+
 if __name__ == '__main__':
     pytest.main()

